packages/openid4vc/src/openid4vc-holder/OpenId4vcSiopHolderService.ts

```diff
@@ -109,6 +110,14 @@ export class OpenId4VcSiopHolderService {
           challenge: nonce,
           domain: clientId,
           presentationSubmissionLocation: DifPresentationExchangeSubmissionLocation.EXTERNAL,
+          openid4vp: {
+            clientId,
+            verifierGeneratedNonce: nonce,
+            mdocGeneratedNonce: await agentContext.wallet.generateNonce(),
```

Comment:
we need to make sure to reuse this nonce when encrypting the resposne (we generate a new nonce there at the moment, but these should be the same)